app/controllers/auth/local_controller.rb

```diff
@@ -12,15 +12,26 @@ def new
 
     def create
       if user = User.find_by_email(params[:session][:email])
-        if BCrypt::Password.new(user.password_hash).is_password? params[:session][:password]
-          context.user = user
-          redirect_back_or_default(root_url)
-          return
+        if user.oauth2_identities and params[:session][:password].nil?
```

Comment:
This conditional is incorrect because it is checking the incoming parameters for a password instead of checking the database for a password_hash. Only local accounts will have a password_hash in the database.